marlowe-playground-client/test/Marlowe/ContractTests.purs

```diff
@@ -20,91 +27,262 @@ import Examples.PureScript.EscrowWithCollateral as EscrowWithCollateral
+--       so that we can have a single definition of contracts and flows, and then test what we care in each one. In semantic
+--       test we care that the compute transaction of term and semantic are the same, in here we care about the output of the simulation.
+escrowSimpleFlow :: TestSuite
+escrowSimpleFlow =
+  test "Escrow" do
+    -- A simple test that runs the Escrow contract to completion
+    let
+      deposit = IDeposit seller buyer ada (BigInteger.fromInt 450)
+
+      choice1 = IChoice ((ChoiceId "Report problem") buyer) (BigInteger.fromInt 1)
+
+      choice2 = IChoice ((ChoiceId "Confirm problem") seller) (BigInteger.fromInt 1)
+
+      finalState =
+        (flip execState mkState) do
+          startSimulation zero filledEscrow
+          applyInput deposit
+          applyInput choice1
+          applyInput choice2
+
+      finalContract = previewOn finalState _currentContract
+
+      txError = do
+        executionState <- preview (_marloweState <<< _Head <<< _executionState <<< _SimulationRunning) finalState
+        executionState ^. _transactionError
+    equal Nothing txError
+    equal (Just Close) (fromTerm =<< finalContract)
+    pure unit
+
+--
+exampleContractsDontHaveErrors :: TestSuite
+exampleContractsDontHaveErrors =
+  suite "Provided Examples don't throw errors nor have warnings" do
+    contractDontHaveErrors "Simple Escrow" filledEscrow
+    contractDontHaveErrors "Escrow with collateral" filledEscrowWithCollateral
+    contractDontHaveErrors "Zero coupon bond" filledZeroCouponBond
+    contractDontHaveErrors "Coupon bond guaranteed" filledCouponBondGuaranteed
+    contractDontHaveErrors "Swap" filledSwap
+    contractDontHaveErrors "Contract for differences" filledContractForDifferences
+    contractDontHaveErrors "Contract for differences with oracle" filledContractForDifferencesWithOracle
+
+-- This is a property based test that checks that for a given contract, the possible actions available
+-- during the simulation don't throw errors nor warnings.
+contractDontHaveErrors :: String -> Term T.Contract -> TestSuite
+contractDontHaveErrors contractName contract =
+  test contractName
+    $ quickCheck
+    $ evalStateT property mkState
+  where
+  property :: StateT Simulation.State Gen Result
+  property = do
+    startSimulation zero contract
+    runContract (Just contract)
+
+  -- When the possible action is a Choose, we need to generate a random value inside the provided Bound.
+  -- We use this function (which looses precision) in order to select a small integer and wrap it in a BigInteger
+  looselyToInt :: BigInteger -> Int
+  looselyToInt = round <<< BigInteger.toNumber
+
+  genValueInBound :: forall m. MonadGen m => MonadRec m => Bound -> m BigInteger
+  genValueInBound (Bound from to) = BigInteger.fromInt <$> chooseInt (looselyToInt from) (looselyToInt to)
+
+  runContract :: Maybe (Term T.Contract) -> StateT Simulation.State Gen Result
+  runContract Nothing = pure Success
```

Comment:
Why `Maybe`? What would `Nothing` mean?

Reply:
Well, it's more of a convenience as the `preview` function returns Maybe. But it's true that if we get Nothing we are doing something wrong. I changed it to `Failed` and the property still holds :)